lib/base/RequestClient.ts

```diff
@@ -92,13 +95,79 @@ export interface RequestClientOptions {
    * The private CA certificate bundle (if private SSL certificate)
    */
   ca?: string | Buffer;
+  /**
+   * Enable auto-retry with exponential backoff when receiving 429 Errors from
+   * the API. Disabled by default.
+   */
+  autoRetry?: boolean;
+  /**
+   * Maximum retry delay in milliseconds for 429 Error response retries.
+   * Defaults to 3000.
+   */
+  maxRetryDelay?: number;
+  /**
+   * Maximum number of request retries for 429 Error responses. Defaults to 3.
+   */
+  maxRetries?: number;
+}
+
+interface BackoffAxiosRequestConfig extends AxiosRequestConfig {
+  /**
+   * Current retry attempt performed by Axios
+   */
+  retryCount?: number;
+}
+
+interface ExpontentialBackoffResponseHandlerOptions {
+  /**
+   * Maximum retry delay in milliseconds
+   */
+  maxIntervalMillis: number;
+  /**
+   * Maximum number of request retries
+   */
+  maxRetries: number;
```

Comment:
nit: Should really be named `maxAttempts` since the first attempt is included in the counting.

Reply:
The `maxRetries` will set the maximum number of retries that the client will make when encountering `429` error responses, not including the first attempt.

The response handler function returned by `getExpontentialBackoffResponseHandler` will run on each response received by the RequestClient's axios instance. Although `retryCount` is incremented in [L146](https://github.com/twilio/twilio-node/pull/886/files#diff-c0ee1ee5484bd0fc609af970b02a1d5a19b8bf82fef04d2b09e76c6374383327R146), the function will evaluate if `retryCount <= maxRetries` before setting the `retryCount` value in the `AxiosRequestConfig` and retrying the request in the conditional block. If the `retryCount` exceeds the `maxRetries`, it will skip the conditional block and return the 429 error response.

Reply:
Ack, my mistake.